### PR TITLE
feat(1958): Upgrade to @hapi/hapi v19.x.x. BREAKING CHANGE: new major data-schema

### PR DESCRIPTION
**Pull request by pritamstyz4ever**
## Context

Screwdriver hapi.js dependencies are outdated and not supported anymore.

## Objective

This PR majorly upgrades @hapi/hapi version and other changes to be compatible with latest @hapi js dependencies.

## References

https://github.com/screwdriver-cd/screwdriver/issues/1958

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -34,48 +34,52 @@ const CUSTOM_PLUGINS = [
...
+            }
+        });
 
-    await server.start();
+        await server.start();
 
-    return server;
+        return server;
+    } catch (err) {
+        throw err;
```
Comment (adong) [inline lib/server.js:83]: `try catch block` would prevent pod from crashing. Do we need to log the errors, or k8s takes care of it?

File: test/plugins/commands.test.js [hunk 2]
```diff
@@ -22,15 +22,15 @@ describe('commands plugin test', () => {
...
         // eslint-disable-next-line global-require
         plugin = require('../../plugins/commands');
 
         server = Hapi.server({
             cache: {
-                engine: catmemory,
-                maxByteSize: 512,
-                allowMixedContent: true
+                engine: new Catmemory({
+                    maxByteSize: 512
```
Comment (adong) [inline test/plugins/commands.test.js:32]: Is `allowMixedContent: true` implicit now?

File: test/plugins/builds.test.js [hunk 3]
```diff
@@ -107,10 +109,12 @@ describe('builds plugin test', () => {
                     headers: {
                         'x-foo': 'bar'
                     },
-                    credentials: {
-                        username: mockBuildID,
-                        scope: ['user']
-                    },
+                    auth: {
+                        strategy: 'token',
+                        credentials: {
```
Comment (adong) [inline test/plugins/builds.test.js:114]: Is `credentials` is part of `auth` now, and we need to explicitly state which `strategy`? 🤔

Comment (pritamstyz4ever) [reply test/plugins/builds.test.js:114]: Yes, thats, correct

File: test/plugins/commands.test.js (hunk 2 above)
Comment (pritamstyz4ever) [reply test/plugins/commands.test.js:32]: Yes, its no longer available. 
https://hapi.dev/module/catbox-memory/api/?v=5.0.0#options

File: lib/server.js (hunk 1 above)
Comment (pritamstyz4ever) [reply lib/server.js:83]: Error is thrown upward and logged.
Its here. https://github.com/screwdriver-cd/store/blob/master/bin/server#L42

File: package.json [hunk 4]
```diff
@@ -59,27 +59,27 @@
...
-    "joi": "13.1.2",
+    "hapi-auth-jwt2": "^10.1.0",
+    "hapi-swagger": "^13.0.2",
+    "joi": "^17.2.0",
     "mime-types": "^2.1.25",
     "request": "^2.88.0",
-    "screwdriver-data-schema": "^18.34.2",
-    "screwdriver-logger": "^1.0.0",
-    "vision": "^5.4.3"
+    "screwdriver-data-schema": "git://github.com/screwdriver-cd/data-schema.git#hapi-v19",
```
Comment (tkyi) [inline package.json:82]: need to update

File: test/plugins/builds.test.js [hunk 5]
```diff
@@ -2,10 +2,10 @@
 
 const { assert } = require('chai');
 const sinon = require('sinon');
-const Hapi = require('hapi');
+const Hapi = require('@hapi/hapi');
 const mockery = require('mockery');
-const catmemory = require('catbox-memory');
-const Boom = require('boom');
+const Catmemory = require('@hapi/catbox-memory');
```
Comment (tkyi) [inline test/plugins/builds.test.js:7]: CatMemory or CatboxMemory might be better for easier readability

File: test/plugins/commands.test.js [hunk 6]
```diff
@@ -22,15 +22,15 @@ describe('commands plugin test', () => {
...
+    beforeEach(async () => {
         // eslint-disable-next-line global-require
         plugin = require('../../plugins/commands');
 
         server = Hapi.server({
             cache: {
-                engine: catmemory,
-                maxByteSize: 512,
-                allowMixedContent: true
+                engine: new Catmemory({
```
Comment (tkyi) [inline test/plugins/commands.test.js:31]: same small comment
might be easier to read as CatboxMemory or CatMemory

File: package.json [hunk 7]
```diff
@@ -59,27 +59,27 @@
...
-    "joi": "13.1.2",
+    "hapi-auth-jwt2": "^10.1.0",
+    "hapi-swagger": "^13.0.2",
+    "joi": "^17.2.0",
     "mime-types": "^2.1.25",
     "request": "^2.88.0",
-    "screwdriver-data-schema": "^18.34.2",
-    "screwdriver-logger": "^1.0.0",
-    "vision": "^5.4.3"
+    "screwdriver-data-schema": "20.0.0",
```
Comment (tkyi) [inline package.json:82]:
```suggestion
    "screwdriver-data-schema": "^20.0.0",
```

File: lib/server.js (hunk 1 above)
Comment (jithine) [reply lib/server.js:83]: catch to throw is not needed, you can remove the try..catch and let caller handle the error.

**Pull request merged**